src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs

```diff
@@ -139,6 +185,127 @@ public virtual Task InitializeAsync()
             return task;
         }
 
+        /// <inheritdoc />
+        public async Task PruneDatabase(ChainedHeader consensusTip, bool nodeInitializing)
+        {
+            this.logger.LogInformation($"Pruning started.");
+
+            if (nodeInitializing)
+            {
+                if (IsDatabasePruned())
+                    return;
+
+                await this.PrepareDatabaseForPruningAsync(consensusTip);
+            }
+
+            this.CompactDataBase();
+
+            this.logger.LogInformation($"Pruning complete.");
+
+            return;
+        }
+
+        private bool IsDatabasePruned()
+        {
+            if (this.TipHashAndHeight.Height <= this.PrunedTip.Height + PruneBlockMargin)
+            {
+                this.logger.LogDebug("(-):true");
+                return true;
+            }
+            else
+            {
+                this.logger.LogDebug("(-):false");
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Compacts the block and transaction database by recreating the tables without the deleted references.
+        /// </summary>
+        /// <param name="consensusTip">The last fully validated block of the node.</param>
+        private async Task PrepareDatabaseForPruningAsync(ChainedHeader consensusTip)
+        {
+            var upperHeight = this.TipHashAndHeight.Height - PruneBlockMargin;
```

Comment:
Unvar here and everywhere where var policy is violated

Reply:
Fixed 👍